### PR TITLE
fix: do not rlp encode extradata

### PR DESCRIPTION
**Pull request by jtraglia**
Instead of RLP encoding the extradata string, just use the string bytes.

For one, the prefix will show up as a unknown character or a space:

telegram-cloud-photo-size-1-5177298186308529445-y

Also, if a user set extradata (via `--builder.extradata`) as a 32-byte string, it would be accepted but when encoded it would become a 33-byte string and would no longer be valid.

### CODE REVIEW

File: crates/node-core/src/cli/config.rs [hunk 1]
```diff
@@ -86,9 +86,9 @@ pub trait PayloadBuilderConfig {
     /// Block extra data set by the payload builder.
     fn extradata(&self) -> Cow<'_, str>;
 
-    /// Returns the rlp-encoded extradata bytes.
-    fn extradata_rlp_bytes(&self) -> Bytes {
-        alloy_rlp::encode(self.extradata().as_bytes()).into()
+    /// Returns the extradata as bytes.
+    fn extradata_bytes(&self) -> Bytes {
+        self.extradata().as_bytes().to_owned().into()
```
Comment (mattsse) [inline crates/node-core/src/cli/config.rs:91]:
```suggestion
        self.extradata().as_bytes().to_vec().into()
```

**Review by mattsse — CHANGES_REQUESTED** [review]
smol pedantic nit

File: crates/node-core/src/cli/config.rs (hunk 1 above)
Comment (jtraglia) [reply crates/node-core/src/cli/config.rs:91]: Ah that's better. Thank you 🙏

**Review by mattsse — APPROVED** [review]
I don't even remember why we had this as rlp

**Pull request merged**